packages/@okta/vuepress-site/docs/reference/api/iam-roles/index.md

```diff
@@ -0,0 +1,3027 @@
+
+###### Response parameters
+
+
+``` http
+HTTP/1.1 204 No Content
+```
+
+###### Request example
+
+
+```bash
+curl -v -X PUT \
+-H "Accept: application/json" \
+-H "Content-Type: application/json" \
+-H "Authorization: SSWS ${api_token}" \
+"https://${yourOktaDomain}/api/v1/groups/00gsr2IepS8YhHRFf0g3/roles/IFIFAX2BIRGUSTQ/targets/catalog/apps/amazon_aws"
+```
+
+###### Response example
+
+
+``` http
+HTTP/1.1 204 No Content
+```
+
+#### Add an App Instance target to an App administrator role
+
+Assign an admin role to a specific App Instance to grant the admin permission to manage an instance of one App or instances of multiple Apps. App Instances are specific Apps that admins have created in their org. For example, there may be a Salesforce App configured differently for each sales region of a company. When you create an App Instance target, an admin may be assigned to manage only two instances of the configured Salesforce Apps and perhaps assigned to manage an instance of another configured App, such as Workday.
+
+> **Note:** You can target a mixture of both App and App Instance targets, but can't assign permissions to manage all instances of an App and then a subset of that same App. For example, you can't specify that an admin has access to manage all instances of a Salesforce app and then also specific configurations of the Salesforce app.
+
+##### Add an App Instance target to an App administrator role given to a User
+
+<ApiOperation method="put" url="/api/v1/users/${userId}/roles/${roleId}/targets/catalog/apps/${appName}/${appInstanceId}" />
+
+Adds an App Instance target for an `APP_ADMIN` Role assigned to a User
+
+When you add the first App or App Instance target, you reduce the scope of the role assignment. The Role no longer applies to all App targets, but applies only to the specified target.
+
+> **Note:** You can target a mixture of both App and App Instance targets, but can't assign permissions to manage all instances of an App and then a subset of that same App. For example, you can't specify that an admin has access to manage all instances of a Salesforce app and then also specific configurations of the Salesforce app.
+
+###### Request parameters
+
+
+| Parameter     | Description                                                  | Param Type   | DataType   | Required |
+| :----------   | :----------------------------------------------------------- | :----------- | :--------- | :------- |
+| userId        | `id` of a User                                               | URL          | String     | TRUE     |
+| roleId        | `id` of a Role                                               | URL          | String     | TRUE     |
+| appName       | `name` of App target from catalog to scope role assignment   | URL          | String     | TRUE     |
+| appInstanceId | `id` of the App Instance target to scope role assignment     | URL          | String     | TRUE     |
```

Comment:
General comments:
1. role > Role
2. Change `id` to just ID in the description.